### PR TITLE
Bulut/21 mock underlay

### PR DESCRIPTION
**Pull request by bbulgu**
Implemented mock underlay and wrote tests.

Fixed `java.rmi.server.ExportException: internal error: ObjID already in use`

Cleaned up unnecessary remoteExceptions

### CODE REVIEW

File: src/main/java/underlay/rmi/RMIUnderlay.java [hunk 1]
```diff
@@ -36,8 +40,16 @@ public RMIUnderlay(int port) {
...
+    } catch (ExportException ee) {
+      try {
+        System.out.println("error during creation");
+        LocateRegistry.getRegistry(port).rebind("RMIImpl", host);
+      } catch (RemoteException re) {
+        System.err.println("[RMIUnderlay] Error while getting registry at port " + port);
+        re.printStackTrace();
+      }
+    } catch (RemoteException e){
+      System.err.println("[RMIUnderlay] Error while creating registry at port " + port);
```
Comment (thep2p) [inline src/main/java/underlay/rmi/RMIUnderlay.java:52]: Is this necessary?

File: src/test/java/underlay/MockUnderlayTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    /**
+     * When both underlays A and B are up and ready, their exchanged messages are delivered to each other intact.
+     * We test different messages to make sure all of them are sent correctly
+     * We compare the results with the actual values returned by the nodes themselves
+     * @throws RemoteException
+     * @throws FileNotFoundException
+     */
+    @Test
+    void sendMessageTest() throws RemoteException, FileNotFoundException {
```
Comment (thep2p) [inline src/test/java/underlay/MockUnderlayTest.java]: Please add more `assertEqual`s involving the `SkipGraphNode` methods, e.g., `SearchByNumIDRequest`.

File: src/main/java/underlay/rmi/RMIUnderlay.java (hunk 1 above)
Comment (bbulgu) [reply src/main/java/underlay/rmi/RMIUnderlay.java:52]: I added this to fix the `java.rmi.server.ExportException: internal error: ObjID already in use` error. More on that here: https://stackoverflow.com/questions/14982760/is-it-possible-to-manually-check-locateregistry-existing

File: src/test/java/underlay/MockUnderlayTest.java (hunk 2 above)
Comment (bbulgu) [reply src/test/java/underlay/MockUnderlayTest.java]: Implemented in d0c717b428fdc07d6ea9ac45a768739880d91858

File: src/main/java/blockchain/ContractCV.java [hunk 3]
```diff
@@ -20,7 +18,7 @@
     Contract ct = new Contract();
     ContractTransaction tesq = new ContractTransaction();
 
-    public ContractCV(LightChainNode owner) throws RemoteException {
+    public ContractCV(LightChainNode owner) {
```
Comment (thep2p) [inline src/main/java/blockchain/ContractCV.java:21]: 👍

**Review by thep2p — APPROVED** [review]

**Pull request merged**